ait/core/server/handler.py

```diff
@@ -84,3 +85,51 @@ def handle(self, input_data):
             tuple of packet UID and message received by stream
         """
         return pickle.dumps((self._pkt_defn.uid, input_data), 2)
+
+
+class CCSDSPacketHandler(Handler):
+
+    def __init__(self, input_type=None, output_type=None, **kwargs):
+        """
+        Params:
+            input_type:   (optional) Specifies expected input type, used to
+                                     validate handler workflow. Defaults to None.
+            output_type:  (optional) Specifies expected output type, used to
+                                     validate handler workflow. Defaults to None
+            **kwargs:     (required) APID values: packet name pairs
+                          (optional) Secondary header length
+        """
+        super(CCSDSPacketHandler, self).__init__(input_type, output_type)
+        self.packet_types = kwargs['packet_types']
+        ait.core.log.info(self.packet_types)
+        self.packet_secondary_header_length = kwargs.get('packet_secondary_header_length', 0)
+        ait.core.log.info(self.packet_secondary_header_length)
+
+    def handle(self, input_data):
+        """
+        Params:
+            packet:    CCSDS packet
+        Returns:
+            tuple of packet UID and packet data field
+        Raises:
+            ValueError:   If packet is specified but not present in default tlm dict.
+        """
+        packet = bytearray(input_data)
+        packet_apid = int(binascii.hexlify(packet[6:8]), 16) & 0x7FF
+        if packet_apid not in self.packet_types:
+            msg = 'CCSDSPacketHandler: Packet APID {} not present in config'.format(packet_apid)
+            msg += ' Available packet APIDs are {}'.format(self.packet_types.keys())
+            raise ValueError(msg)
```

Comment:
Let's print an error and carry on if we get an invalid APID. We shouldn't pull down the handler for one bad APID (since it's far from impossible that a bit will get flipped somewhere, etc.).